### PR TITLE
Rework docs around code splitting

### LINKED ISSUE #1368: Documentation clarification around code splitting
Looking at this section in under the [code splitting](https://github.com/gaearon/react-hot-loader#code-splitting) header:

> To support any code splitting library RHL uses a special tail update detection logic - if some components would be requested after initial HMR event - Application would be updated yet again to apply possible change. In this case you will see a message React-Hot-Loader: some components were updated out-of-bound. Updating your app to reconcile the changes..

I don't understand what this paragraph means. When does the warning show? Does it mean that something is wrong, or does it just indicate that a component on the other side of a code-splitting boundary has been updated?

I'm happy to submit an update to this section once I feel like I understand it.

### PR DESCRIPTION
**Pull request by tstirrat15**
Fixes #1368 

## Motivation
I was having a hard time grokking the docs around code splitting, so I opened #1386. @theKashey helped me understand what was going on, and I wanted to update and clarify the docs if I could.

## Changes
* Rework docs under code splitting header
* Add section specifically about the tail update warning

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -389,14 +389,17 @@ If you are using `babel-plugin` you might not need to apply `webpack-loader` to
 
 ### Code Splitting
 
-If you want to use Code Splitting + React Hot Loader, the simplest solution is to pick one of our compatible library:
+If you want to use Code Splitting + React Hot Loader, the simplest solution is to pick a library compatible with this one:
 
+<!-- TODO: make sure that these are all currently compatible -->
```
Comment (theKashey) [inline README.md]: Well, should we add `React.lazy`?

File: README.md [hunk 2]
```diff
@@ -389,14 +389,17 @@ If you are using `babel-plugin` you might not need to apply `webpack-loader` to
...
 * [Loadable Components](https://github.com/smooth-code/loadable-components/)
```
Comment (theKashey) [inline README.md]: There is a test in the examples. It's a bit complicated, as long as you can test only one library in a single point of time (or they would help each other to update).
Probably the last version of loadable components is not reload-friendly.

File: README.md [hunk 3]
```diff
@@ -389,14 +389,17 @@ If you are using `babel-plugin` you might not need to apply `webpack-loader` to
...
 * [Imported Component](https://github.com/theKashey/react-imported-component)
 * [React Universal Component](https://github.com/faceyspacey/react-universal-component)
 * [React-Loadable](https://github.com/jamiebuilds/react-loadable)
 
-If you use a non-yet-friendly library, like [react-async-component](github.com/ctrlplusb/react-async-component) you have to mark all your "loaded components" as _hot-exported_:
+If you use a not-yet-friendly library, like [react-async-component](github.com/ctrlplusb/react-async-component),
+or are having problems with hot reloading failing to reload code-split components,
+you can manually mark the components at the code-split boundaries.
```
Comment (theKashey) [inline README.md]: you can manually mark the components as hot (below the code-split boundary)

**Comment by theKashey** [comment]
Looking much better!

**Comment by tstirrat15** [comment]
Sweet. I also tweaked those bits.

**Comment by theKashey** [comment]
👍

**Pull request merged**